pinot-spi/src/main/java/org/apache/pinot/spi/plugin/PluginManager.java

```diff
@@ -156,14 +155,18 @@ private PluginClassLoader createClassLoader(Collection<URL> urlList) {
    * @param <T>
    * @return
    */
-  public <T> T createInstance(String pluginName, String className, Class[] argTypes, Object[] argValues)
-      throws Exception {
+  public <T> T createInstance(String pluginName, String className, Class[] argTypes, Object[] argValues) throws Exception{
```

Comment:
Not sure what a null return from here means. There is no public constructor? Should we even return null? Can we return a new type of exception (ConstructorNotFound), making it more explicit?

Reply:
Good catch. I should let it throw exception